monai/handlers/wandb_handlers.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING, Any, Callable, Sequence
+
+import torch
+
+from monai.config import IgniteInfo
+from monai.utils import is_scalar, min_version, optional_import
+
+Events, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Events")
+
+if TYPE_CHECKING:
+    from ignite.engine import Engine
+else:
+    Engine, _ = optional_import(
+        "ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Engine", as_type="decorator"
+    )
+
+if TYPE_CHECKING:
+    import wandb
+else:
+    wandb, _ = optional_import("wandb")
+
+
+DEFAULT_TAG = "Loss"
+
+
+class WandbStatsHandler:
+    """
+    WandbStatsHandler defines a set of Ignite Event-handlers for all the Weights & Biases logging
+    logic. It can be used for any Ignite Engine(trainer, validator and evaluator) and support both
+    epoch level and iteration level. The expected data source is Ignite ``engine.state.output`` and
+    ``engine.state.metrics``.
+    """
+
+    def __init__(
+        self,
+        iteration_log: bool = True,
+        epoch_log: bool = True,
+        epoch_event_writer: Callable[[Engine], Any] | None = None,
+        epoch_interval: int = 1,
+        iteration_event_writer: Callable[[Engine], Any] | None = None,
+        iteration_interval: int = 1,
+        output_transform: Callable = lambda x: x[0],
+        global_epoch_transform: Callable = lambda x: x,
+        state_attributes: Sequence[str] | None = None,
+        tag_name: str = DEFAULT_TAG,
```

Comment:
I don't see the `sync_tensorboard` feature you described in the PR description, how to use it?

Thanks.

Reply:
Hi @Nic-Ma 
`sync_tensorboard` is a feature that is passed to `wandb.init()`. You can find it in the following Colab notebook:

[![](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1jbzl1HlfuLw3lzTWBd2n0m3AKt14eS2-?usp=sharing)

Reply:
OK, I see.
If you want to ensure this feature, maybe you can add some unit tests? Otherwise, we can ignore it.

Thanks.